App/Services/PhotoUtils.js

```diff
@@ -1,94 +1,72 @@
-      if (!thumbExists) {
-        await RNFS.mkdir(thumbDir)
-      }
-      for (const photo of photos) {
-        // iOS method
-        if (photo.uri.includes('assets-library://')) {
-          var regex = /[?&]([^=#]+)=([^&#]*)/g, params = {}, match
-          while (match = regex.exec(photo.uri)) {
-            params[match[1]] = match[2]
-          }
-          const path = fullDir + params.id + '.' + 'JPG'
-          await RNFS.copyAssetsFileIOS(photo.uri, path, 0, 0)
-          photo['path'] = path
-          const thumbPath = await resizeImage(photo.path, thumbRelativeDir)
-          photo['thumbPath'] = thumbPath
-        }
-        // Android Method
-        else if (photo.uri.includes('content://media')) {
-          photo['path'] = await IPFS.getFilePath(photo.uri)
-          photo['thumbPath'] = await resizeImage(photo.path, thumbDir)
-        }
+export async function getPage (pageSize, cursor) {
+  let queryParams = { first: pageSize }
+  if (cursor !== null) {
+    queryParams['after'] = cursor
+  }
+  return CameraRoll.getPhotos(queryParams)
+}
+
+export async function getAllPhotos () {
+  let data = []
+  let hasNextPage = true
+  let cursor = null
+  let totalPages = 10 // I wanted to limit the total size of a return array
+  while (hasNextPage && totalPages) {
+    let photos = await getPage(25, cursor)
+    hasNextPage = photos.page_info.has_next_page
+    cursor = photos.page_info.end_cursor
+    data = data.concat(photos.edges.map((photo) => {
+      // todo: hang on to location data
+      return {
+        timestamp: photo.node.timestamp,
+        uri: photo.node.image.uri,
+        width: photo.node.image.width,
+        height: photo.node.image.height
       }
-      const newestPhotoTimestampString = photos[0].timestamp
-      const newestPhotoTimestamp = new Date(newestPhotoTimestampString * 1000)
-      await AsyncStorage.setItem('latestPhotoTimestamp', newestPhotoTimestamp.toISOString())
-    }
+    }))
+    totalPages -= 1
   }
-  return photos
+  return data
 }
 
-export async function getPhoto (cursor) {
-  let queryParams = { first: 1 }
-  if (cursor !== null) {
-    queryParams['after'] = cursor
+export async function scalePhoto (photo) {
+  const fullDir = RNFS.DocumentDirectoryPath + '/images/full/'
+  const fullExists = await RNFS.exists(fullDir)
+  if (!fullExists) {
+    await RNFS.mkdir(fullDir)
+  }
+  const thumbRelativeDir = 'images/thumb/'
+  const thumbDir = RNFS.DocumentDirectoryPath + '/' + thumbRelativeDir
+  const thumbExists = await RNFS.exists(thumbDir)
+  if (!thumbExists) {
+    await RNFS.mkdir(thumbDir)
   }
-  const photosData = await CameraRoll.getPhotos(queryParams)
-  if (photosData.edges.length < 1) {
-    return
+
+  // iOS method
+  if (photo.uri.includes('assets-library://')) {
+    var regex = /[?&]([^=#]+)=([^&#]*)/g, params = {}, match
+    while (match = regex.exec(photo.uri)) {
+      params[match[1]] = match[2]
+    }
+    const path = fullDir + params.id + '.' + params.ext
```

Comment:
Not sure if the merge will take care of this, but `params.ext` needs to be changed to `JPG`

Reply:
probably out of the scope of the PR. i'm guessing we need some full logic there for jpg/gif/png yeah? and whatever android things to expect

Reply:
Well, this react-native-file-system always outputs a jpeg, so it's safe to hard code there I think.

Reply:
done